iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java

```diff
@@ -174,18 +175,37 @@ private void recovery(File parentFile) throws IOException {
   public void insert(TSRecord tsRecord) throws IOException {
     // memory control
     long memUage = MemUtils.getRecordSize(tsRecord);
-    BasicMemController.getInstance().reportUse(this, memUage);
-    // write data
-    workSupport.insert(tsRecord);
-    valueCount++;
-    // check flush
-    memUage = memSize.addAndGet(memUage);
-    if (memUage > memThreshold) {
-      LOGGER.warn("The usage of memory {} in overflow processor {} reaches the threshold {}",
-          MemUtils.bytesCntToStr(memUage), getProcessorName(),
-          MemUtils.bytesCntToStr(memThreshold));
-      flush();
+    UsageLevel usageLevel = BasicMemController.getInstance().acquireUsage(this, memUage);
+    switch (usageLevel) {
+      case SAFE:
+        // write data
+        workSupport.insert(tsRecord);
+        valueCount++;
+        // check flush
+        memUage = memSize.addAndGet(memUage);
+        if (memUage > memThreshold) {
+          if (LOGGER.isWarnEnabled()) {
+            LOGGER.warn("The usage of memory {} in overflow processor {} reaches the threshold {}",
+                MemUtils.bytesCntToStr(memUage), getProcessorName(),
+                MemUtils.bytesCntToStr(memThreshold));
+          }
+          flush();
+        }
+        break;
+      case WARNING:
+        // write data
+        workSupport.insert(tsRecord);
+        valueCount++;
+        // flush
+        memSize.addAndGet(memUage);
+        flush();
+        break;
+      case DANGEROUS:
+        throw new IOException("The insertion is rejected because dangerous memory level hit");
```

Comment:
How about BufferwriteProcessor? and  is this Exception ok? (e.g., what the impact is for  `ConcreteLogReplayer.multiInsert()`

Reply:
BufferWriteProcessor has similar logic. In the currently serialized recovery method,  memory control is not a concern since the main memory pressure comes from concurrent multiple storage groups.